### DISCUSSION TITLE
usage of json variable

### DISCUSSION THREAD

**Discussion by guidoDimasi** [elsa-workflows/elsa-core#5381] [Q&A] [answered]
how can i iterate a json object when creating a workflow using the designer?
the the json could be a simple object or a json array

**Comment by glime-ai** [comment, marked as answer]
To iterate over a JSON object or array when creating a workflow using the Elsa designer, you can use the `ForEach` activity. This activity allows you to loop through each element in a collection, which can be an array or a list of objects. Here's how you can set it up:

1. **Define the JSON Object or Array**: First, you need to have your JSON object or array defined. This could be set as a workflow variable or obtained from an activity's output.

2. **Add a ForEach Activity**: Drag and drop the `ForEach` activity into your workflow designer.

3. **Configure the ForEach Activity**: Set the `Items` property of the `ForEach` activity to the JSON object or array you want to iterate over. If it's a workflow variable, you can use the JavaScript expression to access it, such as `getMyVariable()`.

4. **Add Activities Inside the ForEach Loop**: Inside the `ForEach` activity, you can add other activities that you want to execute for each element of the collection. For example, you can add a `WriteLine` activity to output the current item.

5. **Access the Current Item**: Within the loop, you can access the current item using the `Current` property of the `ForEach` activity. In JavaScript expressions, you can refer to it using `current()`.

6. **Handle JSON Arrays and Objects**: If you're iterating over an array, the `Current` property will represent the current element. If it's a JSON object with key-value pairs, you may need to use JavaScript to access the keys and values.

Here's an example of a JavaScript expression to access a property of the current item in a JSON array:
```javascript
current().propertyName
```

And if you're dealing with a JSON object and need to iterate over its properties, you might need to convert it to an array of key-value pairs first and then iterate over that array.

Remember to check the Elsa documentation and examples for more detailed guidance on structuring your JSON and using the `ForEach` activity.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).
